crates/primitives/src/alloy_compat.rs

```diff
@@ -36,7 +37,13 @@ impl TryFrom<alloy_rpc_types::Block> for Block {
                     .collect(),
                 alloy_rpc_types::BlockTransactions::Hashes(_) |
                 alloy_rpc_types::BlockTransactions::Uncle => {
-                    return Err(ConversionError::MissingFullTransactions)
+                    // alloy deserializes empty blocks into `BlockTransactions::Uncle`, if the tx
+                    // root is the empty root then we can just return an empty vec.
+                    if block.header.transactions_root == EMPTY_TRANSACTIONS {
+                        Ok(vec![])
+                    } else {
+                        Err(ConversionError::MissingFullTransactions)
+                    }
```

Comment:
would be nice if you can move this drive-by bug fix to a separate pr

Reply:
yep, just added https://github.com/paradigmxyz/reth/pull/8600